src/compiler/parser.rs

```diff
@@ -385,21 +516,70 @@ impl ParserType {
         }
     }
 
+    fn parse_binding_identifier(&mut self) -> Identifier {
+        self.create_identifier(self.is_binding_identifier(), None)
+    }
+
     fn parse_identifier(&mut self, diagnostic_message: Option<DiagnosticMessage>) -> Identifier {
         self.create_identifier(self.is_identifier(), diagnostic_message)
     }
 
-    fn is_list_element(&self, kind: ParsingContext) -> bool {
+    fn is_list_element(&mut self, kind: ParsingContext) -> bool {
         match kind {
             ParsingContext::SourceElements => self.is_start_of_statement(),
+            ParsingContext::VariableDeclarations => {
+                self.is_binding_identifier_or_private_identifier_or_pattern()
+            }
             _ => unimplemented!(),
         }
     }
 
-    fn is_list_terminator(&self) -> bool {
+    fn is_list_terminator(&self, kind: ParsingContext) -> bool {
         if self.token() == SyntaxKind::EndOfFileToken {
             return true;
         }
+        match kind {
+            ParsingContext::VariableDeclarations => self.is_variable_declarator_list_terminator(),
+            _ => false,
+        }
+    }
+
+    fn parse_delimited_list<TItem: Into<Node>>(
+        &mut self,
+        kind: ParsingContext,
+        parse_element: fn(&mut ParserType) -> TItem,
+    ) -> NodeArray {
+        let save_parsing_context = self.parsing_context();
+        self.set_parsing_context(self.parsing_context() | kind);
```

Comment:
What does it mean to `|` these things?

Reply:
The `bitflags` crate automatically implements the bitwise operators for your bitflag types so this is "bitwise-or"'ing these two `ParsingContext` bitflag values (so basically the union of those bitflags)